shared/charts/gpii-flowmanager/templates/pod_disruption_budget.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: {{ template "flowmanager.name" . }}
+  labels:
+    app: {{ template "flowmanager.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: preferences
```

Comment:
I think this should be `flowmanager`.

Is there a way that testing could have caught this bug?

Reply:
Thanks that is a good catch, I'll fix this. This could only be caught if we would test the unhappy scenario - ig. trying to do `kubectl drain` while there are not enough healthy nodes to overtake the workload.

I did test this after fixing the issue. After cordoning all nodes, so that workloads can't be rescheduled, drain on the first node went fine, while on the second one it was blocked, as the PodDisruptionBudget was reached:

```console
$ kubectl drain --delete-local-data=true --timeout=60s --ignore-daemonsets=true gke-k8s-cluster-terraform-20190313220-e725543b-znjb
node "gke-k8s-cluster-terraform-20190313220-e725543b-znjb" already cordoned
WARNING: Ignoring DaemonSet-managed pods: calico-node-b6j84, fluentd-gcp-v3.1.1-6cmkh, ip-masq-agent-klwcc, metadata-agent-m9f8z, metadata-proxy-v0.1-bjq4k
pod "istio-pilot-548b59dc84-jrdn4" evicted
pod "tiller-deploy-5d86c96fcf-mklxt" evicted
pod "istio-sidecar-injector-f555db659-rtd44" evicted
pod "istio-citadel-776fb85794-zsjvj" evicted
pod "istio-ingressgateway-7bd7fd57bf-tn7sg" evicted
pod "calico-node-vertical-autoscaler-547d98499d-pw5lw" evicted
WARNING: Ignoring DaemonSet-managed pods: calico-node-b6j84, fluentd-gcp-v3.1.1-6cmkh, ip-masq-agent-klwcc, metadata-agent-m9f8z, metadata-proxy-v0.1-bjq4k
There are pending pods in node "gke-k8s-cluster-terraform-20190313220-e725543b-znjb" when an error occurred: Drain did not complete within 1m0s
pod/flowmanager-797d64f8cd-gl7f7
pod/nginx-ingress-controller-5776579446-tt2jb
pod/nginx-ingress-default-backend-597f469c8-zk5qz
pod/preferences-7f6cc8f758-229zq
error: unable to drain node "gke-k8s-cluster-terraform-20190313220-e725543b-znjb", aborting command...

There are pending nodes to be drained:
 gke-k8s-cluster-terraform-20190313220-e725543b-znjb
error: Drain did not complete within 1m0s
```

You can see that both `flowmanager` and `preferences` were left pending on the node (note: CouchDB was not running on this particular node, but I've tried it on another node and works for CouchDB as well).

Reply:
Cool, this is what I expected the failure mode to look like. Thanks for confirming it.